src/main/java/io/github/phantamanta44/libnine/client/gui/component/impl/GuiComponentTextInput.java

```diff
@@ -125,8 +161,9 @@ public boolean onKeyPress(int keyCode, char typed) {
         if (focused) {
             int currentLength = value.length();
             if (currentLength < textLength && typed >= 32 && typed < 127) {
-                value += typed;
-                updateValidity();
+                onTextCharInput(typed);
+            } else if (keyCode == Keyboard.KEY_DELETE) {
+                value = "";
```

Comment:
this is no good; users expect the delete key to delete the character in front of the cursor. since the cursor cannot be moved here, the delete key must stay nonfunctional. you can perhaps put this behaviour on ctrl+shift+backspace or something

(is it really even necessary? ctrl+backspace should be good enough for most applications, and certainly for a numeric input)

Reply:
this is not necessary, but I didn't know about ctrl backpace